### DISCUSSION TITLE
how to print heap info

### DISCUSSION THREAD

**Discussion by r3turn00** [eteran/edb-debugger#701] [Q&A]
when i use gdb, i could use "p main_arern", "bins" to print heap info. I wonder if edb have the same features.

**Comment by eteran** [comment]
You can give the Heap Analysis plugin a try under the Plugins menu :-)

**Comment by r3turn00** [comment]
I have tried using Heap Analysis plugin, but it seems helpless. It provides very little information and sometimes even doesn't work

**Comment by eteran** [comment]
I suppose the question is what kind of information are you looking for. The plugin should be able to list nodes if a heap exists (look for things like "[heap]" in the memory map. What other kinds of information are you hoping to find?

**Comment by r3turn00** [comment]
I want to find out the address of main_arena, It would be better if it will show fastbins, small bins and unsorted bins like gdbpwn
`pwndbg> p &main_arena
$2 = (struct malloc_state *) 0x7fe132d19b20 <main_arena>
pwndbg> bins
fastbins
0x20: 0x0
0x30: 0x0
0x40: 0x0
0x50: 0x0
0x60: 0x0
0x70: 0x0
0x80: 0x0
unsortedbin
all: 0x0
smallbins
empty
largebins
empty
`

**Comment by eteran** [comment]
It's been a while since I've worked on the plugin, but these these seem like good things to add! I'll see what happens can do.
